### PR TITLE
Ctctraders 755 - LRN page

### PR DESCRIPTION
**Pull request by kevincoatsworth**

### CODE REVIEW

File: conf/views/localReferenceNumber.njk [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      <div class="govuk-grid-column-two-thirds">
+
+        {% if form.errors.length > 0 %}
+          {{ govukErrorSummary({
+            "titleText": messages("error.summary.title"),
+            "errorList": form.errors
+          }) }}
+        {% endif %}
+
+        <form action="{{ routes.controllers.LocalReferenceNumberController.onSubmit(mode).url }}" method="post">
```
Comment (amaalali) [inline conf/views/localReferenceNumber.njk:30]: We should make a change here and in the templates to pass this as part of the view-model / json payload passed to the view

File: app/pages/LocalReferenceNumberPage.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages
+
+import play.api.libs.json.JsPath
+
+case object LocalReferenceNumberPage extends QuestionPage[String] {
```
Comment (amaalali) [inline app/pages/LocalReferenceNumberPage.scala:21]: Can we save this as a type instead?

File: app/controllers/LocalReferenceNumberController.scala [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        "mode" -> mode
+      )
+
+      renderer.render("localReferenceNumber.njk", json).map(Ok(_))
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData).async {
+    implicit request =>
+
+      val ua = request.userAnswers.getOrElse(UserAnswers(request.internalId))
```
Comment (amaalali) [inline app/controllers/LocalReferenceNumberController.scala:70]: Using internalId isn't correct long term, but is it that it will be addressed in another PR or should we do that here?

File: app/pages/LocalReferenceNumberPage.scala (hunk 2 above)
Comment (garethgooch) [reply app/pages/LocalReferenceNumberPage.scala:21]: We intend to do a type here but this is going to be picked up before update the useranswers and routes

File: conf/views/localReferenceNumber.njk (hunk 1 above)
Comment (garethgooch) [reply conf/views/localReferenceNumber.njk:30]: can we discuss?

File: app/controllers/LocalReferenceNumberController.scala (hunk 3 above)
Comment (garethgooch) [reply app/controllers/LocalReferenceNumberController.scala:70]: Another PR will address this
